#include "adddevice_dialog.h"
#include "ui_adddevice_dialog.h"
#include <vector>
#include <JsonParser.h>
AddDevice_dialog::AddDevice_dialog(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::AddDevice_dialog)
{
    ui->setupUi(this);
    mw = (MainWindow*)parent;
}

AddDevice_dialog::~AddDevice_dialog()
{
    delete ui;
}
void AddDevice_dialog::on_add_device_button_clicked()
{
    QString path = "devices.json";
    auto tmp_text = JsonFile::ReadFile(path);
    QJsonDocument doc = QJsonDocument::fromJson(tmp_text);

    QJsonObject doc_object;

    if(doc.isObject())
    {
        doc_object = doc.object();
    }
    else {
        doc_object = QJsonObject();
    }
    // test

    //auto test = doc_object.value("test").isObject();
    //QString(QJsonDocument(new_object).toJson(QJsonDocument::Indented))
    QJsonObject new_object;
    new_object.insert("os_name", ui->OS_version_comboBox->currentText());
    new_object.insert("description", ui->description_textEdit->toPlainText());
    new_object.insert("ip_adress", ui->ip_lineEdit->text());
    new_object.insert("login", ui->login_lineEdit->text());
    //new_object.insert("password", ui->password_lineEdit->text());
    QJsonValue value(new_object);
    doc_object.insert(ui->name_lineEdit->text(),value);
    //doc.setObject(new_object);



    QJsonDocument res;
    res.setObject(doc_object);
    JsonFile::WriteFile(path, res.toJson());
    //qDebug() << QString(doc.toJson(QJsonDocument::Indented));
    this->parentWidget()->show();
    this->close();

    //delete this;
}


void AddDevice_dialog::on_cancel_button_clicked()
{
    this->parentWidget()->show();
    this->close();
    this->parent();
}


void AddDevice_dialog::on_AddDevice_dialog_finished(int result)
{
    this->parentWidget()->show();
    this->close();
    mw->ShowDevices();
}

